clippy_lints/src/matches/match_ref_pats.rs

```diff
@@ -17,6 +18,11 @@ where
         return;
     }
 
+    // `todo!()` and `unimplemented!()` stub cannot deref
+    if is_todo_unimplemented_stub(cx, scrutinee) {
```

Comment:
The more general problem is that the expression is of type `!`, right? As in, the same issue also happens with `match panic!()`, `match loop {}`, etc.

Can we just check for the never type instead of specifically `todo!()` and `unimplemented!`? That seems more robust
```suggestion
    if cx.typeck_results().expr_ty(scrutinee).is_never() {
```

Reply:
Updated. Thank you!